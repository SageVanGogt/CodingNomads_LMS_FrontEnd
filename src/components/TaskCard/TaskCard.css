.Task_card {
  width: 250px;
  height: 200px;
  margin: 20px;
  background-color: #CECECE; }

.Task_top {
  display: flex;
  justify-content: space-between; }

.Task_edit-btn {
  position: absolute;
  visibility: hidden; }

.Task_card:hover .Task_edit-btn {
  visibility: visible; }
