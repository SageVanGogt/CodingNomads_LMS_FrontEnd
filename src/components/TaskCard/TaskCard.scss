.Task_card {
  width: 250px;
  height: 200px;
  margin: 20px;
  background-color: white;
  font-family: 'Open Sans', sans-serif;
  border-radius: 5px;
  box-shadow: 0 5px 6px -4px rgba(0,0,0,0.22), 0 10px 13px -4px rgba(0,0,0,0.34), inset 0 30px 30px 0 rgba(255,255,255,0.35), inset 0 -30px 30px 0 rgba(0,0,0,0.10);
}

.Task_top {
  display: flex;
  justify-content: space-between;
}

.Task_edit-btn {
  position: relative;
  background-color: #4885a2;
  border-radius: 5px;
  height: 20px;
  width: 40px;
  margin: 10px;
  font-weight: 600;
  color: white;
}

.Task_name {
  margin: 10px;
  font-weight: 800;
}

.Task_desc {
  margin-left: 10px;
  text-align: left;
}

