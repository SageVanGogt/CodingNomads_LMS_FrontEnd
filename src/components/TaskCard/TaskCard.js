import React, { Component} from 'react';
import { connect } from 'react-redux';
import './TaskCard.css';
import PropTypes from 'prop-types';
import { updateCurrentTask } from '../../actions/currentTask';
import { TASKS_EDIT } from '../../constants/routes';
import { withRouter } from 'react-router-dom';

export class TaskCard extends Component {
  handleSelectTaskToEdit = () => {
    const {
      id,
      name,
      description,
      videoLink,
      docs,
      labs,
      updateCurrentTask,
      history } = this.props;
    const currentTask = {
      id,
      name,
      description,
      videoLink,
      docs,
      labs
    };
    updateCurrentTask(currentTask);
    history.push(TASKS_EDIT);
  }

  editBtn = () => {
    return (
      <button
        className="Task_edit-btn"
        onClick={this.handleSelectTaskToEdit}
      >
        edit
      </button>
    );
  }

  render() {
    const { name, description } = this.props;
    const adminEditBtn = this.editBtn();

    return (
      <div className="Task_card">
        <div className="Task_top">
          <h3 className="Task_name">{name}</h3>
          {adminEditBtn}
        </div>
        <hr/>
        <p className="Task_desc">
          {description}
        </p>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  updateCurrentTask: (currentTask) => dispatch(updateCurrentTask(currentTask))
});

TaskCard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  description: PropTypes.string,
  videoLink: PropTypes.string,
  docs: PropTypes.array,
  labs: PropTypes.array,
  user: PropTypes.object,
  updateCurrentTask: PropTypes.func,
  history: PropTypes.object
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(TaskCard)
);
