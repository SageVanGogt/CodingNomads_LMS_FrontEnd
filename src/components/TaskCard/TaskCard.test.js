import React from 'react';
import { shallow } from 'enzyme';
import { TaskCard, mapStateToProps } from './TaskCard';

describe('TaskCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TaskCard />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return a state of the user', () => {
      let mockState = {
        tasks: [{}, {}],
        user: {id: 1, roleId: 1}
      };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.user;
      let expected = {id: 1, roleId: 1};

      expect(actual).toEqual(expected);
    })
  })
})