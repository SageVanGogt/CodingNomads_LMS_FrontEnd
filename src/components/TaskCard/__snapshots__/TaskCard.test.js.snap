// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskCard
    user={
        Object {
            "id": 1,
            "roleId": 2,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="Task_name"
/>,
        <button
          className="Task_edit-btn"
>
          view
</button>,
        <hr />,
        <p
          className="Task_desc"
/>,
      ],
      "className": "Task_card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Task_name",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "view",
          "className": "Task_edit-btn",
        },
        "ref": null,
        "rendered": "view",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Task_desc",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="Task_name"
/>,
          <button
            className="Task_edit-btn"
>
            view
</button>,
          <hr />,
          <p
            className="Task_desc"
/>,
        ],
        "className": "Task_card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Task_name",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "view",
            "className": "Task_edit-btn",
          },
          "ref": null,
          "rendered": "view",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Task_desc",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TaskCard should match the snapshot for admin permissions 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskCard
    user={
        Object {
            "id": 1,
            "roleId": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="Task_name"
/>,
        <button
          className="Task_edit-btn"
          onClick={[Function]}
>
          edit
</button>,
        <hr />,
        <p
          className="Task_desc"
/>,
      ],
      "className": "Task_card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Task_name",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "edit",
          "className": "Task_edit-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "edit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Task_desc",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="Task_name"
/>,
          <button
            className="Task_edit-btn"
            onClick={[Function]}
>
            edit
</button>,
          <hr />,
          <p
            className="Task_desc"
/>,
        ],
        "className": "Task_card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Task_name",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "edit",
            "className": "Task_edit-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "edit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Task_desc",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
